### ISSUE TITLE
Balloon is dropped unintentionally while pressing down right arrow key

### ISSUE THREAD

**Issue by terracoda** [phetsims/balloons-and-static-electricity#314]
When using BASE with the keyboard, my grabbed balloon is dropped unintentionally when holding the right arrow key or down arrow key for distance about the width of the balloon.

It does not happen with WASD keys, or with the the left and up arrow keys.

Using BASE version [1.4.0 dev.4](http://www.colorado.edu/physics/phet/dev/html/balloons-and-static-electricity/1.4.0-dev.4/balloons-and-static-electricity_en.html?accessibility&showGrid) with Safari 10.0.3 and Mac OS Sierra 10.12.3

**Comment by jessegreenberg** [comment]
I haven't run into this yet, Ill see if I can reproduce.  Was VoiceOver in use (or recently in use and turned off)?

**Comment by terracoda** [comment]
@jessegreenberg, Voice Over was not on. I was just using the keyboard alone. I do go back and forth between using VO and then the keyboard alone, but I usually reload the sim often.

It seems to happen if two balloons are active or if two balloons have been active recently. Maybe it has something to do with two balloon activation?

**Comment by jessegreenberg** [comment]
@terracoda how often does this happen?

> I do go back and forth between using VO and then the keyboard alone, but I usually reload the sim often.

VoiceOver has a bug where if you close it without restarting Safari, the OS will still  do weird things with keyboard events as if VoiceOver was still on, we saw this in john-travoltage. My initial guess is that this is happening. I am going to grab a OS machine to test now.

**Comment by terracoda** [comment]
@jessegreenberg, you are absolutely right! It's VoiceOver and Mac OS causing this problem. I had not restarted VO between reloads of the sim.

It only happens after I use VoiceOver.

**Comment by terracoda** [comment]
Re
> I had not restarted VO between reloads of the sim.

I meant to say: I had not restarted Safari in between reloads.

**Comment by jessegreenberg** [comment]
Great @terracoda, good to know. I also tried without VO on macOS 10.12.1 Safari and i haven't been able to get it to happen yet.

**Comment by terracoda** [comment]
If this is a know issue, how should we mark this issue?

**Comment by jessegreenberg** [comment]
With VoiceOver on in 1.4.0-dev.4 the arrow keys aren't working at ALL for me, do they  work  at all for you with VoiceOVer on @terracoda?

**Comment by terracoda** [comment]
I'm using 10.12.3 and I haven't tried upgrading since the last I tried, I spent 3 days on the phone with AppleCare restoring my system. 

It's possible that 10.12.3 might be the problem?

**Comment by terracoda** [comment]
The left and right arrow keys can get a little sluggish, but they do generally work with VO on. WASD keys work consistently as designed, so they are the better option.

**Comment by jessegreenberg** [comment]
Possible, it looks like 10.12.1 is also very problematic.

**Comment by terracoda** [comment]
Why do platform OSs even versions of OSs have to differ so much????

**Comment by jessegreenberg** [comment]
Arrggh, I had "quick nav" on, and VO was keeping that setting toggled when turning it off. Turning quick  nav off and the arrow keys work again.

**Comment by terracoda** [comment]
Hmm, with Quick Nav off, it is still broken for me.
Did you still have o reload Safari after turning quick nav off?

**Comment by jessegreenberg** [comment]
Sorry, what is broken? I thought the arrow keys were generally working for you, do you mean it is releasing after pressing and holding?

**Comment by jessegreenberg** [comment]
Just verified that pressing and holding is broken for me too when VO is on. :(

**Comment by jessegreenberg** [comment]
It is happening consistently, every time.

**Comment by terracoda** [comment]
Yes, releasing after pressing and holding, but still working.

**Comment by jessegreenberg** [comment]
@terracoda in user settings Accessibility/Keyboard -> Keyboard Preferences -> Key Repeat

If I turn off Key Repeat the problem goes away.  Can you please confirm?

**Comment by jessegreenberg** [comment]
Similarly, if I slide the "Delay Until Repeat" slider all the way to "short", I cant drag the balloon as far before encountering the problem.

**Comment by jessegreenberg** [comment]
Watching events, I can see that the balloon is released when a second 'keydown' event comes through because of key repeat. For some reason, VO doesn't like this, but *only for the right arrow key*.

**Comment by jessegreenberg** [comment]
Aha! The KeyboardDragHandler has this code:

```js
      // if the key is already down, don't do anything else (we don't want to create a new keystate object
      // for a key that is already being tracked and down, nor call startDrag every keydown event)
      if ( self.keyInListDown( [ event.keyCode ] ) ) { return; }

      // required to work with Safari and VoiceOver, otherwise arrow keys will move virtual cursor
      if ( Input.isArrowKey( event.keyCode ) ) {
        event.preventDefault();
      }
```

We need to reverse this order so that ```preventDefault`` can be called always before returning out.

**Comment by jessegreenberg** [comment]
Should be fixed in the above commit, but @terracoda it would be great for you to verify in a version, Ill paste next dev version here and assign to you once it is ready.

**Comment by jessegreenberg** [comment]
@terracoda would you please verify that this is no longer a problem on your computer with VoiceOver?

**Comment by terracoda** [comment]
It is no longer an issue in dev version 7.
I only lose the balloon if Quick Nav goes on.
